### PR TITLE
Ensure that arrays of Classes are handled as strings in ComponentProp…

### LINKED ISSUE #2698: @ExportedService annotation (DS 1.4) not processed correctly by bnd 4.1.0
When trying to use the newly added @ExportedService annotation from DS 1.4 with bnd-maven-plugin 4.1.0 in order to declare the service.exported.interfaces property as follows:

```
@Component(service = TestService.class)
@ExportedService(service_exported_interfaces = TestService.class)
public class TestServiceImpl implements TestService {}
```

Results in the following error being reported by bnd:

```
[ERROR] Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:4.1.0:bnd-process (default) on project test-project: Invalid data type ConcreteRef -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:4.1.0:bnd-process (default) on project test-project: Invalid data type ConcreteRef
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoFailureException: Invalid data type ConcreteRef
    at aQute.bnd.maven.plugin.BndMavenPlugin.reportErrorsAndWarnings (BndMavenPlugin.java:468)
    at aQute.bnd.maven.plugin.BndMavenPlugin.execute (BndMavenPlugin.java:328)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

### PR DESCRIPTION
**Pull request by timothyjward**
…ertyType annotations

Fixes #2698

Signed-off-by: Tim Ward <timothyjward@apache.org>

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotationReader.java [hunk 1]
```diff
@@ -633,7 +633,9 @@ private void processActivationObject(String propertyDefKey, ClassTypeSignature p
 			for (Entry<String, Object> entry : componentPropertyAnnotation.entrySet()) {
 				String key = entry.getKey();
 				Object value = entry.getValue();
-				handleValue(key, value, value instanceof TypeRef, null);
+				boolean isClass = value instanceof TypeRef || (value.getClass()
+					.isArray() && ((Object[]) value)[0] instanceof TypeRef);
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotationReader.java]: I think you also need to handle the case where the array is empty. e.g. `@ClassArrayProp({})`. So the `isClass` test should be like:

```
boolean isClass = value instanceof TypeRef || (value.getClass()
	.isArray() && (Array.getLength(value) > 0) && (Array.get(value, 0) instanceof TypeRef));
```

Perhaps another test with an empty array can be added too.

**Comment by timothyjward** [comment]
I have added the test and updated the `isClass` calculation. 

@bjhargrave Just to confirm, our expectation is that an empty array will result in no properties in the component xml?

**Comment by bjhargrave** [comment]
> @bjhargrave Just to confirm, our expectation is that an empty array will result in no properties in the component xml?

Yes, see [112.8.3](https://osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-ordering.generated.properties) bullet 2:

> For component property type methods with a value of an empty array, a property element must not be generated.
